Assets/Scripts/Controllers/MouseController.cs

```diff
@@ -220,18 +220,19 @@ void RebuildSelectionStuffInTile()
     {
 
         // Make sure stuffInTile is big enough to handle all the characters, plus the 3 extra values
-        mySelection.stuffInTile = new ISelectable[ mySelection.tile.characters.Count + 3 ];
+        mySelection.stuffInTile = new List<ISelectable>( );
 
         // Copy the character references
-        for (int i = 0; i < mySelection.tile.characters.Count; i++)
+        foreach (Character character in mySelection.tile.characters)
         {
-            mySelection.stuffInTile[i] = mySelection.tile.characters[i];
+            mySelection.stuffInTile.Add(character);
         }
 
         // Now assign references to the other three sub-selections available
-        mySelection.stuffInTile[mySelection.stuffInTile.Length - 3] = mySelection.tile.furniture;
-        mySelection.stuffInTile[mySelection.stuffInTile.Length - 2] = mySelection.tile.inventory;
-        mySelection.stuffInTile[mySelection.stuffInTile.Length - 1] = mySelection.tile;
+        mySelection.stuffInTile.Add( mySelection.tile.furniture );
+        mySelection.stuffInTile.Add( mySelection.tile.inventory );
+        mySelection.stuffInTile.Add( mySelection.tile );
+        mySelection.stuffInTile.Add( mySelection.tile.pendingBuildJob );
```

Comment:
I think it can go any way, I personally would put it on top of all three, but I'll leave it to tweaks later on.